modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/store/FilePageStoreManager.java

```diff
@@ -500,4 +505,32 @@ public void destroyGroupIfExists(int groupId) throws IOException {
     private Path groupDir(int groupId) {
         return dbDir.resolve(GROUP_DIR_PREFIX + groupId);
     }
+
+    /**
+     * Scans the working directory to find IDs of all groups for which directories still remain.
+     *
+     * @return IDs of groups.
+     */
+    public Set<Integer> allGroupIdsOnFs() {
+        try (Stream<Path> tableDirs = Files.find(dbDir, 1, isTableDir())) {
+            return tableDirs.map(FilePageStoreManager::extractTableId).collect(toUnmodifiableSet());
+        } catch (IOException e) {
+            throw new IgniteInternalException(Common.INTERNAL_ERR, "Cannot scan for groupIDs", e);
+        }
+    }
+
+    private BiPredicate<Path, BasicFileAttributes> isTableDir() {
+        return (path, attrs) -> {
+            return attrs.isDirectory() && !dbDir.startsWith(path) && path.getFileName().toString().startsWith(GROUP_DIR_PREFIX);
```

Comment:
What does `!dbDir.startsWith(path)` do here? I don't think I understand it

Reply:
Rewrote it with `Files.list()` which does not need this